### PR TITLE
feat: implement xpack security user api

### PR DESCRIPTION
**Pull request by jfroche**
Implements:

  * Create/Update User
  * Delete User
  * Get User
  * Disable User
  * Enable User

### CODE REVIEW

**Comment by jfroche** [comment]
refs #1113

**Comment by olivere** [comment]
Looking great. Thank you! Will merge in 7.0.8.

File: xpack_security_get_user.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+// XPackSecurityUser is the user object.
+//
+// The Java source for this struct is defined here:
+// https://github.com/elastic/elasticsearch/blob/7.3/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/User.java
+type XPackSecurityUser struct {
+	Username     string                 `json:"username"`
+	Roles        []string               `json:"roles"`
+	Fullname     string                 `json:"full_name"`
+	Email        string                 `json:"email"`
+	Password     string                 `json:"password"`
```
Comment (ckaye) [inline xpack_security_get_user.go]: I don't think the `Password` and `PasswordHash` are returned in the GetUser API? docs here: https://www.elastic.co/guide/en/elasticsearch/reference/7.3/security-api-get-user.html

Comment (olivere) [reply xpack_security_get_user.go]: Correct. It seems like [these fields will be serialized](https://github.com/elastic/elasticsearch/blob/7.4/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/User.java#L165-L175).

File: xpack_security_get_user.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	"net/url"
+
+	"github.com/olivere/elastic/v7/uritemplates"
+)
+
+// XPackSecurityGetUserService retrieves a user by its name.
+// See https://www.elastic.co/guide/en/elasticsearch/reference/7.0/security-api-get-user.html.
+type XPackSecurityGetUserService struct {
+	client *Client
+	pretty bool
```
Comment (olivere) [inline xpack_security_get_user.go]: The other standard fields are missing in this API as well, e.g. `filter_path`, `header` etc.

File: xpack_security_get_user.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	name   string
```
Comment (olivere) [inline xpack_security_get_user.go]: The `name` is not a valid parameter for this API. See the [REST API spec](https://github.com/elastic/elasticsearch/blob/7.4/x-pack/plugin/src/test/resources/rest-api-spec/api/security.get_user.json) for a list of parameters: It should be one or more `username`.

**Review by olivere — CHANGES_REQUESTED** [review]
Can you please check that all services have all [common fields](https://github.com/elastic/elasticsearch/blob/7.4/rest-api-spec/src/main/resources/rest-api-spec/api/_common.json) (except `source`) and [the official XPack REST API spec](https://github.com/elastic/elasticsearch/tree/7.4/x-pack/plugin/src/test/resources/rest-api-spec/api) matches the parameters in the Go services in this PR.

File: xpack_security_get_user.go (hunk 2 above)
Comment (jfroche) [reply xpack_security_get_user.go]: Thank you for the review,
I have added the common options (based on https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html)

File: xpack_security_get_user.go (hunk 3 above)
Comment (jfroche) [reply xpack_security_get_user.go]: Right, I have added username as a list of strings. Are you ok with that ?

File: xpack_security_get_user.go (hunk 1 above)
Comment (jfroche) [reply xpack_security_get_user.go]: Indeed, well seen ! Thank you

**Comment by olivere** [comment]
I changed a few things like aligning strongly to the [REST API Spec](https://github.com/elastic/elasticsearch/tree/master/x-pack/plugin/src/test/resources/rest-api-spec/api), keeping names the same everywhere (e.g. `Username`), and adding missing common options.

**Comment by phillbaker** [comment]
@olivere would you accept a PR backporting this to v6?

**Pull request merged**